plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,177 +82,391 @@ function executeCLI($scriptName, &$args) {
-				exit();
-			}
+		$file = new SplFileObject($filename, 'r');
+		$file->setFlags(SplFileObject::READ_CSV);
 
-			$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-			$authorDao = DAORegistry::getDAO('AuthorDAO'); /* @var $authorDao AuthorDAO */
+		$file->rewind();
+
+		if ($file->eof()) {
+			echo __('plugins.importexport.csv.fileDoesNotExist', ['filename' => $filename]) . "\n";
+			exit;
+		}
+
+		if (!$file->eof()) {
+			// Imports
+			import('lib.pkp.classes.submission.SubmissionFile'); // constants.
+			import('lib.pkp.classes.file.FileManager');
+			import('lib.pkp.classes.core.Core');
+			import('classes.file.PublicFileManager');
+
+			// DAOs
 			$pressDao = Application::getContextDAO();
+			$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
 			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+			$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
+			$publicationDao = DAORegistry::getDAO('PublicationDAO'); /* @var $publicationDao PublicationDAO */
 			$seriesDao = DAORegistry::getDAO('SeriesDAO'); /* @var $seriesDao SeriesDAO */
+			$authorDao = DAORegistry::getDAO('AuthorDAO'); /* @var $authorDao AuthorDAO */
 			$publicationFormatDao = DAORegistry::getDAO('PublicationFormatDAO'); /* @var $publicationFormatDao PublicationFormatDAO */
-			$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
-			import('lib.pkp.classes.submission.SubmissionFile'); // constants.
-			$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
 			$publicationDateDao = DAORegistry::getDAO('PublicationDateDAO'); /* @var $publicationDateDao PublicationDateDAO */
+			$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+			$categoryDao = DAORegistry::getDAO('CategoryDAO'); /* @var $categoryDao CategoryDAO */
+
+			// Static variables
+			$fileManager = new FileManager();
+			$publicFileManager = new PublicFileManager();
+			$dirNames = Application::getFileDirectories();
+			$format = trim($dirNames['context'], '/') . '/%d/' . trim($dirNames['submission'], '/') . '/%d';
+			$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+
+			/** @var \PKP\services\PKPFileService */
+			$fileService = Services::get('file');
+			/** @var \PKP\services\PKPPublicationService */
+			$publicationService = Services::get('publication');
+
+			$invalidRows = [];
+
+			// Cache variables
+			$genres = [];
+			$authorGroups = [];
+			$series = [];
+			$cachedCategories = [];
+
+			$headers = $file->fgets(); // Retrieve headers in case of invalid rows;
+
+			while(!$file->eof()) {
+				$position = $file->ftell();
+				$rowString = $file->fgets();
+
+				$file->fseek($position);
+
+				$fields = $file->fgetcsv();
```

Comment:
Now that I saw the rest of the code... I'll leave a second comment 😁

As a user, I think it's interesting to have a file just with the rows that failed and the reason. But the code should be optimized.

Instead of storing all rows in memory (`$invalidRows`), you can create the error file on the beginning and append the errors there with `$errorFile->fputcsv()`, if no errors happened, then erase the file.

Also, if you keep the same structure of the input (just with an extra column to show the error), it will be easier for the user to fix the problems and re-run the import, basically `$errorFile->fputcsv(array_merge(array_pad($fields, 15, null), $error))`.